### ISSUE TITLE
fresh dataset

### ISSUE THREAD

**Issue by dhecloud** [endernewton/tf-faster-rcnn#140]
How do I train a fresh model using my own synthetic dataset(about 90k images) from scratch? I don't want to use the pretrained VGG16 weights. 

I read somewhere I can use a Xavier initialiser to initialise the starting weights, but how do I go about doing it?

**Comment by lhoangan** [comment]
The pre-trained weight file is processed in the train_model function [(here)](https://github.com/endernewton/tf-faster-rcnn/blob/master/lib/model/train_val.py#L162) of `lib/model/train_val.py`. 

Basically, the whole graph is initialized with [`sess.run(tf.variables_initializer(variables, name='init'))`](https://github.com/endernewton/tf-faster-rcnn/blob/master/lib/model/train_val.py#L161) before the pre-trained weights are loaded.

So if you don't pass the pre-trained weight using `--weight` flag to the train script `tools/trainval_net.py`, you can check if `self.pretrained_model` is None and bypass the whole block of code that does the weight loading (until this [line](https://github.com/endernewton/tf-faster-rcnn/blob/master/lib/model/train_val.py#L173)

**Comment by endernewton** [comment]
thanks @lhoangan

**Comment by jtdowdall** [comment]
When I follow this method for resnet101, I am getting a NaN loss:

`iter: 20 / 70000, total loss: nan
 >>> rpn_loss_cls: nan
 >>> rpn_loss_box: nan
 >>> loss_cls: 0.641103
 >>> loss_box: 0.000000
 >>> lr: 0.001000`

Which quickly errors out. I think it may have to do with the weight initialization but I can't figure it out. Any help?

**Comment by dby2017** [comment]
> When I follow this method for resnet101, I am getting a NaN loss:
> 
> `iter: 20 / 70000, total loss: nan
> 
> > > > rpn_loss_cls: nan
> > > > rpn_loss_box: nan
> > > > loss_cls: 0.641103
> > > > loss_box: 0.000000
> > > > lr: 0.001000`
> 
> Which quickly errors out. I think it may have to do with the weight initialization but I can't figure it out. Any help?

Do you solve this problem?

**Comment by jtdowdall** [comment]
If I recall correctly, it had to do with bounding boxes with coordinates outside of the image dimensions.
